### PR TITLE
Add a new yaml provider, allows providing yaml bytes

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: config/configmapprovider/bytes_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+func TestBytesProvider_InvalidValue(t *testing.T) {
+	sp := NewBytes()
+	_, err := sp.Retrieve(context.Background(), "bytes::2s", nil)
+	assert.Error(t, err)
+}
+
+func TestBytesProvider(t *testing.T) {
+	sp := NewBytes()
+
+	ret, err := sp.Retrieve(context.Background(), "bytes:processors::batch::timeout: 2s", nil)
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply config/configmapprovider/bytes_test.go:39]: I also kind of thought the same but was lazy to change after initial implementation. Good that we have an agreement here

File: config/configmapprovider/bytes.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+//   bytes-location   = "bytes:" yaml-bytes
+//
+// Examples:
+// `bytes:processors::batch::timeout: 2s`
+// `bytes:processors::batch/foo::timeout: 3s`
+func NewBytes() Provider {
+	return &bytesMapProvider{}
+}
+
+func (s *bytesMapProvider) Retrieve(_ context.Context, location string, _ WatcherFunc) (Retrieved, error) {
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply config/configmapprovider/bytes.go:43]: I will open an issue to discuss this, I don't think this is a blocker on this PR, correct?

[the remark this answers is not in the document]
Comment (mx-psi) [reply config/configmapprovider/bytes.go:43]: Yeah, not a blocker :)

File: config/configmapprovider/bytes_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply config/configmapprovider/bytes_test.go]: Done.

File: config/configmapprovider/bytes_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	assert.Error(t, err)
+}
+
+func TestBytesProvider_InvalidValue(t *testing.T) {
+	sp := NewBytes()
+	_, err := sp.Retrieve(context.Background(), "bytes::2s", nil)
+	assert.Error(t, err)
+}
+
+func TestBytesProvider(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply config/configmapprovider/bytes_test.go]: Done.

**Review by mx-psi — APPROVED** [review]

File: config/configmapprovider/bytes.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply config/configmapprovider/bytes.go]: Issue filed.

File: config/configmapprovider/yaml.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+// NewYaml returns a new Provider that allows to provide yaml bytes.
+//
+// This Provider supports "yaml" scheme, and can be called with a "location" that follows:
+//   bytes-location   = "yaml:" yaml-bytes
+//
+// Examples:
+// `yaml:processors::batch::timeout: 2s`
+// `yaml:processors::batch/foo::timeout: 3s`
+func NewYaml() Provider {
```
Comment (jpkrohling) [inline config/configmapprovider/yaml.go:39]: Shouldn't this be `NewYAML` per Go conventions?

**Review by Aneurysm9 — APPROVED** [review]

File: config/configmapprovider/yaml.go (hunk 4 above)
Comment (bogdandrutu) [reply config/configmapprovider/yaml.go:39]: I wish this was a lint error.

Comment (bogdandrutu) [reply config/configmapprovider/yaml.go]: Done.

**Pull request merged**